ross-nichols-solutions.js

```diff
@@ -0,0 +1,154 @@
+      return true
+    } else if (str.includes('e')){
+      return true
+    } else if (str.includes('i')) {
+      return true
+    } else if (str.includes('o')) {
+      return true
+    } else if (str.includes('u')) {
+      return true
+    } else if (str.includes('y')) {
+      return 'sometimes'
+    } else {
+      return false
+    }
+  }
+  
+  isAVowel('a')
+  console.log(isAVowel('a'))
+  
+  //#3
+  const getTwoLengths = (str1, str2) => {
+    const arr = []
+    length1 = str1.length
+    length2 = str2.length
+    arr.push(length1)
+    arr.push(length2)
+    return arr
+  }
+  
+  getTwoLengths('Hank', 'Hippopopalous')
+  console.log(getTwoLengths('Hank', 'Hippopopalous'))
+  
+  //#4
+  const getMultipleLengths = (arr) => {
+    let wordCount = []
+    for(i = 0; i < arr.length; i++) {
+      len = arr[i].length
+      //console.log(len)
+      wordCount.push(len)
+    }
+    console.log(wordCount)
+  }
+  
+  getMultipleLengths(['hello','what','is','up','dude'])
+  
+  
+  
+  //PART TWO
+  //#1
+  const maxOfThree = (num1, num2, num3) => {
+    return Math.max(num1, num2, num3)
+  }
+  
+  maxOfThree(6, 9, 1)
+  console.log(maxOfThree(6, 9, 1))
+  
+  //#2
+  const printLongestWord = (arr) => {
+    let longest = arr[0]
+    for (let i = 1; i < arr.length; i++) {
+      if (longest.length <= arr[i].length) {
+        longest = arr[i]
+      }
+      //console.log(longest)
+    }
+    return longest
+  }
+  
+  printLongestWord(["BoJack", "Princess", "Diane", "a", "Max", "Peanutbutter", "big", "blob"])
+  console.log(printLongestWord(["BoJack", "Princess", "Diane", "a", "Max", "Peanutbutter", "big", "blob"]))
+  
+  //#3
+  const transmogrify = (num1, num2, num3) => {
+    //multiply first two numbers, raise to power of third
+    return (num1 * num2) ** num3
+  }
+  
+  console.log(transmogrify(5, 3, 2))
+  
+  //#4
+  //Part 1: Write a function to return fibonacci sequence up to number
+  //Sum of number is equal to sum of previous two numbers -> (n-1) + (n-2)
+  //Part 2: Push even numbers into array
+  //Part 3: Return summed value of array
+  //Part 4: Find sum of even numbers less or equal four million in array
+  const removeOdd = (arr) => {
+    let newArray = []
+    let newTotal = 0
+    for(let i = 0; i < arr.length; i++) {
+      if(arr[i] % 2 === 0) {
+        newArray.push(arr[i])
+        newTotal += arr[i]
+      }
+    }
+    if(newTotal >= 4000000) {
+      return 4000000
+    }
+    console.log(newArray)
+    console.log(newTotal)
+    return newArray
+    return newTotal
+  }
+  
+  const euler = (num) => {
+    let fibArray = [0, 1]
+    for(let i = 2; i <= num; i++) {
+      fibArray[i] = fibArray[i - 1] + fibArray[i - 2]
+    }
+    removeOdd(fibArray)
+  }
+  
+  euler(10)
+  
+  
+  //#5 find the needle
+  //for loop to search for index = 'needle'?
+  const findNeedle = (arr) => {
+    for(i = 0; i < arr.length; i++) {
+      if(arr[i] = 'needle') {
+        indexValue = arr[i].length - 1
```

Comment:
this is a really funny coincidence! What you're saying in this line of code is that the `indexValue` is the value of the length of the element at index value `i` in array `arr` (minus 1). In other words, the `indexValue` is equal to the length of the word `needle`, 6, minus 1, to get us 5.

It just so happens that the array that we gave you puts the word needle at index value 5! But that's not what your code says here. To test it, try using the array `['hay', 'needle', 'junk', 'moreJunk']`; you'll see the output still says that it found the needle at position 5.

The logical step I would like you to make when working with for loops over arrays (which will happen a bunch) is that the `i` in your for loop is the value you are using to match up with the index value of the element! So, in this case, `indexValue = i`, or you could just use `i` in place of the variable `indexValue`. Does that make sense? Feel free to reach out if you want to talk this through a bit more.